packages/neo-one-client-core/src/user/RemoteReceiveUserAccountProvider.ts

```diff
@@ -0,0 +1,70 @@
+import { Block, RawAction, UserAccountProvider } from '@neo-one/client-common';
+import { Endpoint, Message } from './messageTypes';
+
+export interface RemoteReceiveUserAccountProviderOptions {
+  readonly endpoint: Endpoint;
+  readonly userAccountProvider: UserAccountProvider;
+}
+
+export class RemoteReceiveUserAccountProvider {
+  private readonly messageEndpoint: Endpoint;
+  private readonly userAccountProvider: UserAccountProvider;
+  // tslint:disable-next-line readonly-keyword
+  private readonly asyncIterableInstanceMap: { [id: string]: AsyncIterator<Block | RawAction> };
+
+  public constructor({ endpoint, userAccountProvider }: RemoteReceiveUserAccountProviderOptions) {
+    this.messageEndpoint = endpoint;
+    this.userAccountProvider = userAccountProvider;
+    this.asyncIterableInstanceMap = {};
+
+    this.messageEndpoint.addEventListener(async (event: Message) => {
+      if (event.type === 'METHOD') {
+        try {
+          // tslint:disable-next-line no-any
+          if ((this.userAccountProvider as any)[event.method] === undefined) {
+            throw new Error(`Method ${event.method} is not implemented by the requested Provider.`);
+          }
+          // tslint:disable-next-line no-any
+          const value = await (this.userAccountProvider as any)[event.method](...event.args);
```

Comment:
Cast once, i.e. `const userAccountProvider: any = userAccountProviderIn` and remove all of the duplicate casting.